switchbot_ros/scripts/switchbot_ros_server.py

```diff
@@ -24,32 +26,69 @@ def __init__(self):
                 self.token = f.read().replace('\n', '')
         else:
             self.token = token
-        try:
-            self.bots = SwitchBotAPIClient(token=self.token)
-            device_list_str = 'Switchbot device list:\n'
-            device_list = sorted(
-                self.bots.device_list,
-                key=lambda device: str(device['deviceName']))
-            for device in device_list:
-                device_list_str += 'Name: ' + str(device['deviceName'])
-                device_list_str += ', Type: ' + str(device['deviceType'])
-                device_list_str += '\n'
-            rospy.loginfo(device_list_str)
-        except ConnectionError as e: # If the machine is not connected to the internet
-            self.bots = None
-            rospy.logwarn('Failed to connect to the switchbot server. The client would try connecting to it when subscribes the ActionGoal topic.')
+        # Initialize switchbot client
+        self.bots = self.get_switchbot_client()
+        self.print_devices()
         # Actionlib
         self._as = actionlib.SimpleActionServer(
             '~switch', SwitchBotCommandAction,
             execute_cb=self.execute_cb, auto_start=False)
         self._as.start()
+        # Topic
+        self.pub = rospy.Publisher('~devices', DeviceArray, queue_size=1)
```

Comment:
Could you make the publisher `latch=True` like
https://github.com/PR2/app_manager/blob/3391ee3f54b4e7a1a05695d9db78a463d71ce315/src/app_manager/app_manager.py#L145-L147 ?